lib/CAPI/Dialect/FIRRTL.cpp

```diff
@@ -4,9 +4,158 @@
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.h"
 #include "mlir/CAPI/IR.h"
 #include "mlir/CAPI/Registration.h"
 #include "mlir/CAPI/Support.h"
 
+using namespace circt::firrtl;
+
+//===----------------------------------------------------------------------===//
+// Dialect API.
+//===----------------------------------------------------------------------===//
+
 MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(FIRRTL, firrtl,
                                       circt::firrtl::FIRRTLDialect)
+
+//===----------------------------------------------------------------------===//
+// Type API.
+//===----------------------------------------------------------------------===//
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFIRRTLType(MlirType type) {
+  return unwrap(type).isa<FIRRTLType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsGround(MlirType type) {
+  return unwrap(type).cast<FIRRTLType>().isGround();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsConst(MlirType type) {
+  return unwrap(type).cast<FIRRTLType>().isConst();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsClock(MlirType type) {
+  return unwrap(type).isa<ClockType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsReset(MlirType type) {
+  return unwrap(type).isa<ResetType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAsyncReset(MlirType type) {
+  return unwrap(type).isa<AsyncResetType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsSInt(MlirType type) {
+  return unwrap(type).isa<SIntType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsUInt(MlirType type) {
+  return unwrap(type).isa<UIntType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAnalog(MlirType type) {
+  return unwrap(type).isa<AnalogType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsBundle(MlirType type) {
+  return unwrap(type).isa<BundleType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFVector(MlirType type) {
+  return unwrap(type).isa<FVectorType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenBundle(MlirType type) {
+  return unwrap(type).isa<OpenBundleType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenVector(MlirType type) {
+  return unwrap(type).isa<OpenVectorType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFEnum(MlirType type) {
+  return unwrap(type).isa<FEnumType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypeIsRef(MlirType type) {
+  return unwrap(type).isa<RefType>();
+}
+
+MLIR_CAPI_EXPORTED bool firrtlTypesAreEquivalent(MlirType dest, MlirType src,
+                                                 bool srcOuterTypeIsConst) {
+  return circt::firrtl::areTypesEquivalent(unwrap(dest).cast<FIRRTLType>(),
+                                           unwrap(src).cast<FIRRTLType>(),
+                                           srcOuterTypeIsConst);
+}
+
+MLIR_CAPI_EXPORTED int32_t firrtlTypeGetBitWidth(MlirType type,
+                                                 bool ignoreFlip) {
+  return getBitWidth(unwrap(type).cast<FIRRTLBaseType>(), ignoreFlip)
+      .value_or(0);
```

Comment:
In Chisel side, unknown widths are specified as `-1`, IMO here should be `-1` as well.